### PR TITLE
Replace Icons in WebUI

### PR DESCRIPTION
**Pull request by rudream**
## Purpose

This PR resolves https://github.com/gravitational/cloud/issues/4279

This PR completely replaces our current icon library with Phosphor icons.

`e` counterpart: https://github.com/gravitational/teleport.e/pull/1859

## Implementation

SVG icons are converted to React components via a script that can be run with `yarn process-icons`.

The vast majority of the additions in this PR are from the generated files and snapshot updates. The relevant changes to be reviewed are primarily in `design/Icon`, alongside many minor adjustments to fix sizing and alignment throughout the app.

### CODE REVIEW

File: web/packages/design/src/Icon/Icon.tsx
Comment (ravicious) [inline web/packages/design/src/Icon/Icon.tsx:1]: For some reason, the icons are rendered differently in Firefox vs Chrome & Safari.

Take a look at the CCStripe icon.

| Chrome | Firefox |
| --- | --- |
| chrome | Firefox |

As you can see, in Chrome the icon is only 3px "higher" above the first full white portion of the letter C. In Firefox, it's 4px. This makes the icons feel slightly misaligned in Firefox.

At first I ran into this when inspecting the story for the search bar in Connect (which so far is rendered only in Chromium anyway), but then I realized that the problem persists in the design/Icon story as well.

[I dumped my screenshots into Figma](https://www.figma.com/file/6hsDMvZkiwlZCSdp4SlhFd/Untitled?type=design&node-id=0%3A1&mode=design&t=Td2kI8IP3ph9oyqs-1) if you'd like to compare them more closely.

File: web/packages/design/src/Icon/Icons/GitHub.tsx
Comment (ravicious) [inline web/packages/design/src/Icon/Icons/GitHub.tsx:1]: This looks like a regular cat. 😔

File: web/packages/design/src/Icon/Icon.tsx
Comment (ravicious) [inline web/packages/design/src/Icon/Icon.tsx:1]: On a 1080p monitor, some of the icons look very poorly in Chrome compared to Firefox, they're all pixelized for some reason.

Chrome vs Firefox 1080p

Even on Retina, some icons in Chrome look considerably worse than in Firefox, for example the Warning icon.

Chrome vs Firefox Retina

File: web/packages/teleterm/src/ui/StatusBar/ShareFeedback/ShareFeedback.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ export function ShareFeedback() {
         onClick={openShareFeedback}
       >
         {!hasBeenShareFeedbackOpened && <NotOpenedYetIndicator />}
-        <ChatBubble fontSize="14px" />
+        <ChatBubble size="medium" />
```
Comment (ravicious) [inline web/packages/teleterm/src/ui/StatusBar/ShareFeedback/ShareFeedback.tsx:50]: This icon is in the bottom right of the window. idk if it was always like that, but in the new design the icon looks misaligned.

File: web/packages/design/src/Icon/Icons/Cross.tsx
Comment (ravicious) [inline web/packages/design/src/Icon/Icons/Cross.tsx:1]: The cross looks quite weird under the light theme on my 1080p display. ;f

cross

**Review by ravicious — COMMENTED** [review]
I looked through the teleterm changes and the code changes themselves looks fine. The biggest problem is that the icons are weirdly pixelized in Chrome.

File: web/packages/teleterm/src/ui/StatusBar/ShareFeedback/ShareFeedback.tsx (hunk 1 above)
Comment (rudream) [reply web/packages/teleterm/src/ui/StatusBar/ShareFeedback/ShareFeedback.tsx:50]: added a small margin below it, how's this?

### Before

image

### After

image

File: web/packages/design/src/Icon/Icon.tsx
Comment (rudream) [reply web/packages/design/src/Icon/Icon.tsx:1]: I'm not really sure what to do about this as these are differences in the browsers' ways of rendering. If something only looks misaligned in Connect, then we can just manually adjust it there by adding some extra margin, but for the browser app, trying to fix misalignment in one browser would just make it look misaligned in another browser. Do you have any ideas?

Comment (rudream) [reply web/packages/design/src/Icon/Icon.tsx:1]: We noticed this and looked into it but after trying out various methods, we couldn't find a way to resolve it and ultimately decided we should move on with the way they are even if some icons appear slightly pixelated on Chrome. This is a well documented issue, Chrome struggles with rendering diagonal lines in SVG's smoothly (unless they're perfectly 45°). Also, the previous icon library had similar issues (look at `Auth Connectors` and `Session Recordings` especially):

image

cc @roraback

Comment (ravicious) [reply web/packages/design/src/Icon/Icon.tsx:1]: I have no idea what might be causing this unfortunately, I'd have hoped that someone on the internet has documented this. ;f If anything, I'd suspect that it has to do with how we style the icon with

```css
  display: inline-flex;
  align-items: center;
  justify-content: center;
```

Maybe there are differences in how Firefox and Chrome handle SVGs in flex children.

**Review by ravicious — APPROVED** [review]
I'm traveling, so I don't even have a way to test the 1080p changes. But on Retina the icons look good and we can always iron out any rough edges in the future.

It sounds like you and Kenny have already spent a considerable amount of work trying to make them look as good as possible.

File: web/packages/teleport/src/Audit/EventList/EventTypeCell.tsx [hunk 2]
```diff
@@ -225,12 +225,11 @@ const EventIconMap: Record<EventCode, React.FC> = {
 };
 
 export default function renderTypeCell(event: Event, clusterId: string) {
-  const IconType = EventIconMap[event.code] || Icons.List;
+  const Icon = EventIconMap[event.code] || Icons.ListThin;
 
   const iconProps = {
-    p: '1',
-    mr: '3',
-    fontSize: '3',
```
Comment (kimlisa) [inline web/packages/teleport/src/Audit/EventList/EventTypeCell.tsx:233]: the audit log screen has icons that are more bigger. it's too big imo, it increases vertical spacing and decreases width for some column.

take a look, icons are not the most important thing to display, so i think we can decrease the size? i'm using macbook pro 16 inch full screen

image

File: web/packages/design/src/SVGIcon/index.ts [hunk 3]
```diff
@@ -69,3 +69,6 @@ export { UpgradeIcon } from './Upgrade';
 export { UserIcon } from './User';
 export { UsersIcon } from './Users';
 export { UserSettingsIcon } from './UserSettings';
+
+export { SVGIcon } from './SVGIcon';
```
Comment (kimlisa) [inline web/packages/design/src/SVGIcon/index.ts:73]: I'm getting confused.

I know we want to use phosphor icons now. ✅ 

But we also have svg icons that are not part of phosphor icons. I looked at storybook for our icons and we have like duplicates. for example "users", the phosphor icon and... "non-phosphor" icon looks different?

File: web/packages/design/src/Icon/Icons.story.tsx [hunk 4]
```diff
@@ -16,54 +16,57 @@ limitations under the License.
...
 };
 
-export const DeprecatedIcons = () => (
+/*
+
+THIS FILE IS GENERATED. DO NOT EDIT.
+
+*/
+
+export const Icons = () => (
```
Comment (kimlisa) [inline web/packages/design/src/Icon/Icons.story.tsx:35]:
```suggestion
export const PhosphorIcons = () => (
```

b/c we have a story called "PreferSvgIcon". We should change that to mean like... non phosphor icons?

**Review by kimlisa — COMMENTED** [review]
thanks for your patience.

i am loving the new loading indicator!! and the new icons look 🤌 

also idk what the original looked like, but is it just me that the info icon with text next to it looks more out of alignment than usual 😅 

image


for the sidenav, the previous version had "users" that used "users icon". i see phosphor has users icon, we should use that instead?

old -> new (image)
image

File: web/packages/design/src/SVGIcon/index.ts (hunk 3 above)
Comment (kimlisa) [reply web/packages/design/src/SVGIcon/index.ts:73]: i guess what i'm getting at is, should we clean it up and have phoshpor icon's be the source of truth?

Comment (rudream) [reply web/packages/design/src/SVGIcon/index.ts:73]: Yes, in a subsequent PR I'll clean up `SVGIcon` and remove duplicates, I didn't want to make this PR and larger than it is and wanted to keep its scope solely to replacing icomoon with Phosphor

File: web/packages/design/src/Icon/Icons.story.tsx (hunk 4 above)
Comment (rudream) [reply web/packages/design/src/Icon/Icons.story.tsx:35]: I think we should keep this as just `Icons` and rename the `PreferSvgIcon` story to something else, maybe `CustomIcons`?

Comment (kimlisa) [reply web/packages/design/src/Icon/Icons.story.tsx:35]: yeah lets do that 👍

**Review by zmb3 — APPROVED** [review]

**Review by kimlisa — APPROVED** [review]

**Pull request merged**